### PR TITLE
chore!: Rename `servicePath` to `basePath`

### PR DESCRIPTION
**Pull request by tomfrenken**
With this PR, servicePath and all its associated functions (e.g. getServicePath, setCustomServicePath, ...) are renamed to their basePath equivalent.

Relates to SAP/cloud-sdk-backlog#790.

- [x] I know which base branch I chose for this PR, as the default branch is `v2-main` now, which is not for v3 development.
- [ ] If my change will be merged into the `main` branch (for v3), I've updated (V3-Upgrade-Guide.md)[./V3-Upgrade-Guide.md] in case my change has any implications for users updating to SDK v3

### CODE REVIEW

File: .changeset/new-brooms-punch.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+'@sap-cloud-sdk/http-client': major
+'@sap-cloud-sdk/resilience': major
+'@sap-cloud-sdk/generator': major
+'@sap-cloud-sdk/odata-v2': major
+'@sap-cloud-sdk/odata-v4': major
+'@sap-cloud-sdk/openapi': major
+'@sap-cloud-sdk/util': major
+---
+
+Rename servicePath to basePath.
```
Comment (FrankEssenberger) [inline .changeset/new-brooms-punch.md:21]: [req] add the indicator like [New Functionality] here.

File: packages/odata-common/src/request/odata-request-config.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ const logger = createLogger({
  */
 export abstract class ODataRequestConfig {
   payload: Record<string, any> | string;
-  customServicePath: string;
+  basePath: string;
```
Comment (FrankEssenberger) [inline packages/odata-common/src/request/odata-request-config.ts:23]: [q] Should we ceep the custom prefix?

**Review by FrankEssenberger — COMMENTED** [review]
[req] Please also add a section to the migration guide what needs to be adjusted for existing service mappings.

File: packages/odata-common/src/request-builder/request-builder-base.ts [hunk 3]
```diff
@@ -75,12 +75,12 @@ export abstract class MethodRequestBuilder<
...
    * Replace the default service path with the given custom path.
-   * In case of the SAP S/4HANA APIs the servicePath defaults to `/sap/opu/odata/sap/<SERVICE_NAME>` and can be overwritten here.
-   * @param servicePath - Path to override the default with.
+   * In case of the SAP S/4HANA APIs the basePath defaults to `/sap/opu/odata/sap/<SERVICE_NAME>` and can be overwritten here.
+   * @param basePath - Path to override the default with.
    * @returns The request builder itself, to facilitate method chaining.
    */
-  setCustomServicePath(servicePath: string): this {
-    this.requestConfig.customServicePath = servicePath;
+  setBasePath(basePath: string): this {
```
Comment (FrankEssenberger) [inline packages/odata-common/src/request-builder/request-builder-base.ts:82]: [q] Should we keep the custom?

**Review by FrankEssenberger — APPROVED** [review]
Please adjust the changeset and add content to the migration guide. I already approve so that you can merge after these changes. Codewise it was just a renaming. A few question about a `custom` prefix.

File: packages/odata-common/src/request/odata-request-config.ts (hunk 2 above)
Comment (tomfrenken) [reply packages/odata-common/src/request/odata-request-config.ts:23]: I was wondering the same thing, however in the ticket one of the TODOs is:
`Rename the setCustomServicePath parameter for OData and replace it with setBasePath`
Therefore I assumed the removal of the `custom` would be applied here as well.

**Pull request merged**